test/conftest.py

```diff
@@ -230,6 +230,17 @@ def urls(self):
             self.driver.switch_to.window(handle)
             yield self.driver.current_url
 
+    def get_canvas_data(self):
+        import base64
+        canvas_tag_property = "//canvas[starts-with(@id, 'matplotlib')]"
+        canvas_element = self.driver.find_element_by_xpath(canvas_tag_property)
+        img_script = "return arguments[0].toDataURL('image/png').substring(21)"
+        canvas_base64 = self.driver.execute_script(img_script, canvas_element)
+        canvas_png = base64.b64decode(canvas_base64)
+        with open(TEST_PATH /
+                  r"canvas-{0}.png".format(self.browser), 'wb') as f:
+            f.write(canvas_png)
```

Comment:
Is this going to be used outside of the matplotlib test? If not it might be better to move it to the matplotlib file directly as a function, if possible.

Reply:
By the `matplotlib` file, do you mean the new backend aka the file `html5_canvas_backend.py`? This wouldn't be possible since the `backend` does not have access to `selenium` (which is ideal). OR are you talking about the file `test_matplotlib.py` file? This could be done but would require some extra lines that define the variable `TEST_PATH`. What is preferred?

Reply:
I think @rth's suggestion of moving this to the `test_matplotlib.py` file makes sense, given that it is matplotlib-specific.